executor/adapter.go

```diff
@@ -80,6 +81,26 @@ func (a *statement) OriginText() string {
 	return a.text
 }
 
+// isPointGetByPkOrUniqueKey returns true when current plan is point get by pk or unique key
+func (a *statement) isPointGetByPkOrUniqueKey(ctx context.Context) bool {
+	checkPlan := a.plan
+	if proj, ok := checkPlan.(*plan.Projection); ok {
+		if len(proj.Children()) != 1 {
+			return false
+		}
+		checkPlan = proj.Children()[0]
+	}
+
+	if indexScan, ok := checkPlan.(*plan.PhysicalIndexScan); ok {
+		return len(indexScan.Ranges) == 1 && indexScan.Ranges[0].IsPoint(ctx.GetSessionVars().StmtCtx)
```

Comment:
It's not enough to check index range is point.
Need to check index is unique and low value count is the same index columns count.